lib/ossert/fetch/stackoverflow.rb

```diff
@@ -1,67 +1,165 @@
       end
 
-      # Public: Fetch resolved questions
-      #
-      # Returns a response Hash from SO.
-      def fetch_resolved_questions
-        fetch_questions accepted: true
+      # Fetch resolved questions
+      # @param params [Hash<String, String>] special params for the search
+      # @return [Hash<Symbol, Hash>] a response from SO
+      def fetch_resolved_questions(params = {})
+        fetch_questions(params.merge(accepted: true))
+      end
+
+      # Fetch several pages of questions seamlessly
+      # @param params [Hash<String, String>] special params for the search
+      # @yieldparam question [Hash<Symbol, Hash>] one of the fetched questions
+      # @return [nil] Not specified
+      def fetch_questions_batch(params = {})
+        page = 1
+        resp = fetch_questions(params.merge(page: page))
+        while resp[:items].length.positive? && page < MAX_PAGES
+          resp[:items].each { |question| yield question }
+          page += 1
+          resp = fetch_questions(params.merge(page: page))
+        end
       end
 
-      # Public: Fetch & process total stats
+      # Process total stats
       #
-      # Returns nothing.
+      # Process aggregated data for the project on StackOverflow.
+      # @return [nil] Not specified.
       def process_questions_total
-        community.total.questions_count = fetch_questions[:total]
-        community.total.questions_resolved_count = fetch_resolved_questions[:total]
+        community.total.questions_count = fetch_questions(filter: :total)[:total]
+        community.total.questions_resolved_count = fetch_resolved_questions(filter: :total)[:total]
+        if last_question = fetch_questions(order: :desc, pagesize: 5).dig(:items, 0)
+          community.total.last_question_date = last_question[:creation_date]
+        end
+      end
+
+      # Process last year stats
+      #
+      # Process aggregated data for the project on StackOverflow
+      # for the last 5 quarters.
+      # @return [nil] Not specified.
+      def process_last_year_questions
+        start_date = (Time.now.utc - 1.year).beginning_of_quarter.to_i
+
+        fetch_questions_batch(fromdate: start_date) do |question|
+          quarter = community.quarters[question[:creation_date]]
+
+          quarter.questions << question[:question_id]
+          if question[:is_answered]
+            quarter.questions_resolved << question[:question_id]
+          end
+          if question[:owner] && !quarter.questioners.include?(question[:owner][:user_id])
+            quarter.questioners << question[:owner][:user_id]
+            quarter.questioner_rep << question[:owner][:reputation] # TODO: couple with questioner
+          end
+          quarter.question_view << question[:view_count]
+          quarter.question_score << question[:score]
+          quarter.answers << question[:answer_count]
+        end
       end
     end
   end
-end
\ No newline at end of file
+end
+
+# {
```

Comment:
I find it pretty inconvenient to start the file with ~30 lines of useless for most developers information. Nevertheless it is obviously related to the class in general, so I specified it in the class comments. Do you think it would be better to merge it?